plugins/audit-log-node/src/DefaultAuditLogger.ts

```diff
@@ -0,0 +1,159 @@
+        ? credentials.principal.userEntityRef
+        : undefined;
+
+      const serviceEntityRef = this.authService.isPrincipal(
+        credentials,
+        'service',
+      )
+        ? credentials.principal.subject
+        : undefined;
+
+      return userEntityRef ?? serviceEntityRef;
+    } catch {
+      return undefined;
+    }
+  }
+  async createAuditLogDetails(
+    options: AuditLogDetailsOptions,
+  ): Promise<AuditLogDetails> {
+    const { eventName, stage, metadata, request, response, status } = options;
+
+    const actorId = options.actorId || (await this.getActorId(request)) || null;
+
+    // Secrets in the body field should be redacted by the user before passing in the request object
+    const auditRequest = request
+      ? {
+          method: request.method,
+          url: request.originalUrl,
+          params: request.params,
+          query: request.query,
+          body: request.body,
+        }
+      : undefined;
+
+    const actor: ActorDetails = { actorId };
+    if (request) {
+      actor.ip = request.ip;
+      actor.hostname = request.hostname;
+      actor.userAgent = request.get('user-agent');
+    }
+
+    const auditLogCommonDetails = {
+      actor,
+      meta: metadata || {},
+      request: auditRequest,
+      isAuditLog: true as const,
+      response,
+      eventName,
+      stage,
+    };
+
+    if (auditRequest) {
+      auditLogCommonDetails.request = auditRequest;
+    }
+    if (response) {
+      auditLogCommonDetails.response = response;
+    }
+
+    if (status === 'failed') {
+      const errs = options.errors as ErrorLike[];
+      return {
+        ...auditLogCommonDetails,
+        status,
+        errors: errs.map(err => {
+          return {
+            name: err.name,
+            message: err.message,
+            stack: err.stack,
+          };
+        }),
+      };
+    }
+
+    return {
+      ...auditLogCommonDetails,
+      status,
+    };
+  }
+  async auditLog(options: AuditLogOptions): Promise<void> {
+    const logLevel = options.level || 'info';
+    const auditLogDetails = await this.createAuditLogDetails({
+      eventName: options.eventName,
+      status: 'succeeded',
+      stage: options.stage,
+      actorId: options.actorId,
+      request: options.request,
+      response: options.response,
+      metadata: options.metadata,
+    });
+    switch (logLevel) {
+      case 'info':
+        this.logger.info(options.message, auditLogDetails as JsonObject);
+        return;
+      case 'debug':
+        this.logger.debug(options.message, auditLogDetails as JsonObject);
+        return;
+      case 'warn':
+        this.logger.warn(options.message, auditLogDetails as JsonObject);
+        return;
+      case 'error':
+        this.logger.error(options.message, auditLogDetails as JsonObject);
+        return;
+      default:
+        throw new Error(`Log level of ${logLevel} is not supported`);
+    }
```

Comment:
```suggestion
    this.logger.log(logLevel, options.message, auditLogDetails as JsonObject)
```

Reply:
The backstage `LoggerService` does not have the `log` function unforunately.

Reply:
https://github.com/backstage/backstage/blob/b1ecb9f32536a279fd798a58d66e333ea3c4b1f0/packages/backend-plugin-api/src/services/definitions/LoggerService.ts#L24